### PR TITLE
#825 CR Delete Endpoint

### LINKED ISSUE #825: Backend - Make the delete CR endpoint
### Description

we need an endpoint to be able to delete a change request for this epic #314 

### Acceptance Criteria

a new endpoint that can be called that deletes the given change request

(note: check the epic #314 for all business logic and ask questions if the below doesn't line up)

### Proposed Solution

- the new endpoint should be on the route `/:crId/delete` (which means it needs to be placed in the route file right above the review one)
- it does not need any body parameters, as the user submitting this request can be acquired from the auth
- the service function should have the parameters (user: User, crId: number)
- the service function should error if the crid does not exist, if the user is not an admin/app_admin, or if the cr has already been reviewed
- if it's good to delete, change the date deleted and deleted by for the change request in prisma as well as all associated objects as needed

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Created CR Delete Endpoint

## Screenshots
Screen Shot 2023-02-09 at 1 15 32 PM

Screen Shot 2023-02-09 at 1 35 21 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #825  (issue #)

### CODE REVIEW

File: src/backend/src/routes/change-requests.routes.ts [hunk 1]
```diff
@@ -67,6 +67,8 @@ changeRequestsRouter.post(
   ChangeRequestsController.createStandardChangeRequest
 );
 
+changeRequestsRouter.post('/:crid/delete', ChangeRequestsController.deleteChangeRequest);
```
Comment (anthonybernardi) [inline src/backend/src/routes/change-requests.routes.ts:70]: .delete not .post (also you'll need to update CORS, see Nezam's PR)

File: src/backend/src/services/change-requests.services.ts [hunk 2]
```diff
@@ -582,4 +582,26 @@ export default class ChangeRequestsService {
...
+  static async deleteChangeRequest(submitter: User, crId: number): Promise<void> {
+    // verify user is allowed to delete change requests
+    if (!(submitter.role === 'ADMIN' || submitter.role === 'APP_ADMIN')) throw new AccessDeniedException();
+
+    // ensure existence of change request
+    const foundCR = await prisma.change_Request.findUnique({
+      where: { crId }
+    });
+
+    if (!foundCR) throw new NotFoundException('Change Request', crId);
```
Comment (anthonybernardi) [inline src/backend/src/services/change-requests.services.ts:600]: should we throw if it's already deleted?

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice, didn't realize this one was so simple lol

gonna need unit tests and manual tests tho

File: src/backend/src/services/change-requests.services.ts (hunk 2 above)
Comment (Peyton-McKee) [reply src/backend/src/services/change-requests.services.ts:600]: probably i didnt think about that

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
so im realizing this problem where hte actual changes objects dont have deleted fields which means theyre gonna show up on the work packages and projects pages anyways i think

which means we need to change all those query args to include stuff like this:

Screenshot 2023-02-11 at 4 57 00 PM

everything else looks good thoguh!

File: src/backend/tests/change-requests.test.ts [hunk 3]
```diff
@@ -289,6 +289,71 @@ describe('Change Requests', () => {
     });
   });
 
+  describe('add Proposed Solution', () => {
```
Comment (anthonybernardi) [inline src/backend/tests/change-requests.test.ts:292]: looks like some mixing here

File: src/backend/tests/change-requests.test.ts [hunk 4]
```diff
@@ -289,6 +289,71 @@ describe('Change Requests', () => {
...
+    afterEach(() => {
```
Comment (anthonybernardi) [inline src/backend/tests/change-requests.test.ts:293]: only need one after each and it should be outer

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
looks like tests got mixed up in merging stuff

File: src/backend/tests/change-requests.test.ts (hunk 3 above)
Comment (Peyton-McKee) [reply src/backend/tests/change-requests.test.ts:292]: I dont know what github is doing tbh, this is weird

**Review by anthonybernardi — APPROVED** [review]
beautiful

**Pull request merged**